Src/Sarif.PatternMatcher/AnalyzeOptions.cs

```diff
@@ -51,5 +51,16 @@ public class AnalyzeOptions : AnalyzeOptionsBase
             "enhanced-reporting",
             HelpText = "Enable enhanced reporting provided by dynamic validators (when --dynamic-validation is also enabled).")]
         public bool EnhancedReporting { get; internal set; }
+
+        [Option(
+            "retry",
+            HelpText = "Enable retry connection if enabled.")]
+        public bool Retry { get; set; }
+
+        [Option(
+            "max-memory",
+            HelpText = "The maximum memory that can be used for RE2.",
+            Default = -1)]
+        public long MaxMemory { get; internal set; }
```

Comment:
>MaxMemory [](http://example.com/codeflow?start=20&length=9)

You did not make it clear whether you are specifying KB or bytes. It is a good practice to always make this clear. Suggest this:

--max-memory-in-kb

Remember when I gave you this note and you changed the other name in this file to FileSizeInKilobytes? :)

While you're at it, let's change that other argument name to:

--file-size-in-kb

Reply:
fixed :)